ddl/ddl_api.go

```diff
@@ -5499,6 +5502,29 @@ func (d *ddl) DropView(ctx sessionctx.Context, stmt *ast.DropTableStmt) (err err
 	return d.dropTableObject(ctx, stmt.Tables, stmt.IfExists, viewObject)
 }
 
+func checkTableHasForeignKeyReferred(ctx sessionctx.Context, is infoschema.InfoSchema, schema, tbl model.CIStr, ignoreTables []*ast.TableName) *model.ReferredFKInfo {
```

Comment:
Do we need to do this in the background dropping/truncating table?

Reply:
Nice catch, we should check this in DDL owner too, aka double-check.

Reply:
done.